core/rawdb/freezer.go

```diff
@@ -71,14 +72,16 @@ type Freezer struct {
 	tables       map[string]*freezerTable // Data tables for storing everything
 	instanceLock *flock.Flock             // File-system lock to prevent double opens
 	closeOnce    sync.Once
+
+	eradb *eradb.EraDatabase
```

Comment:
I feel it's wrong to put eradb here.

Freezer itself can be used for general purposes, e.g. for hosting chain segment, state data, and so on. 

Era (I feel) is strongly tight with chain segment, at least the initial design goal is for the chain segment before the merge.

I would prefer to move all era logic into chainFreezer

Reply:
That makes sense. It was a bit weird to initialize the eradb only for chain freezer but not for state.